docs/tutorials/http-serving-tutorial.md

```diff
@@ -30,43 +31,82 @@ static async insertGreeting(ctxt: TransactionContext<Knex>, friend: string, note
 Each associates a function with an HTTP URL.
 
-:::info
 
-You might be wondering why we don't talk about setting up an HTTP server.
-It's because DBOS is _serverless_.
-When you run an app locally with `npx dbos start`, we manage the HTTP server for you, using the endpoints and configuration you specify with decorators.
+### Inputs and HTTP Requests
 
-:::
+When a function has arguments other than its context, DBOS automatically parses them from the HTTP request, and returns an error to the client if they are not found.
 
-### Handlers
+Arguments can be parsed from three places:
 
-A function annotated with an endpoint decorator but no other decorators is called a _handler_ and must take a [`HandlerContext`](../api-reference/contexts#handlercontext) as its first argument, like in the first example above.
-Handlers can [invoke](../api-reference/contexts#handlerctxtinvoke) other functions and directly access HTTP requests and responses.
-However, DBOS makes no guarantees about handler execution: if a handler fails, it is not automatically retried.
-You should use handlers when you need to access HTTP requests or responses directly or when you are writing a lightweight task that does not need the strong guarantees of transactions and workflows.
+#### 1. URL Path Parameters
 
-### Inputs and HTTP Requests
+You can include a path parameter placeholder in a URL by prefixing it with a colon, like `name` in this example:
+
+```javascript
+@GetApi('/greeting/:name')
+static async greetingEndpoint(ctx: HandlerContext, name: string) {
+  return `Greeting, ${name}`;
+}
+```
+
+Then, give your method an argument with a matching name (such as `name: string` above) and it is automatically parsed from the path parameter.
+
+For example, if we send our app this request, then our method is called with `name` `dbos`:
+
+```
+GET /greeting/dbos
+```
+
+#### 2. URL Query String Parameters
+
+[`GET`](../api-reference/decorators#getapi) and [`DELETE`](../api-reference/decorators.md#deleteapi) endpoints automatically parse arguments from query strings.
+
+For example, the following endpoint expects the `id` and `name` parameters to be passed through a query string:
 
-When a function has arguments other than its context (e.g., `name`, `friend`, or `note` in the snippets above), DBOS automatically parses them from the HTTP request, and returns an error to the client if they are not found.
+```javascript
+@GetApi('/example')
+static async exampleGet(ctx: HandlerContext, id: number, name: string) {
+  return `${id} and ${name} are parsed from URL query string parameters`;
+}
+```
+
+If we send our app this request, then our method is called with `id` `123` and `name` `dbos`:
+
+```
+GET /example?id=123&name=dbos
+```
+
+#### 3. HTTP Body Fields
+
+[`POST`](../api-reference/decorators#postapi), [`PATCH`](../api-reference/decorators#patchapi), and [`PUT`](../api-reference/decorators#putapi) endpoints automatically parse arguments from the HTTP request body.
```

Comment:
For other types of endpoints, you can use `@ArgSource(ArgSources.BODY)` to explicitly parse an argument from the request body.